x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -246,15 +256,28 @@ private void runJob(JobTask jobTask, JobState jobState, OpenJobAction.JobParams
 
                     // This job has a running datafeed attached to it.
                     // In order to prevent gaps in the model we revert to the current snapshot deleting intervening results.
-                    revertToCurrentSnapshot(jobTask.getJobId(), ActionListener.wrap(response -> openJob(jobTask), jobTask::markAsFailed));
+                    revertToCurrentSnapshot(jobTask.getJobId(), ActionListener.wrap(
+                        response -> openJob(jobTask),
+                        e -> {
+                            if (autodetectProcessManager.isNodeDying() == false) {
+                                jobTask.markAsFailed(e);
+                            }
+                        }
+                    ));
                 } else {
                     openJob(jobTask);
                 }
             },
-            jobTask::markAsFailed
+            e -> {
+                if (autodetectProcessManager.isNodeDying() == false) {
+                    jobTask.markAsFailed(e);
+                }
+            }
         );
 
-        hasRunningDatafeedTask(jobTask.getJobId(), hasRunningDatafeedTaskListener);
+        if (autodetectProcessManager.isNodeDying() == false) {
+            hasRunningDatafeedTask(jobTask.getJobId(), hasRunningDatafeedTaskListener);
+        }
```

Comment:
I personally would like there to be a predicate before `hasRunningDatafeedTaskListener` is even created. Similar to the `FAILED` check. 

```java
if (autodetectProcessManager.isNodeDying()) {
    return;
}
```

But that is not a huge deal, it just reads a little strange.